app/models/article.rb

```diff
@@ -71,6 +81,37 @@ def self.current_page(params)
     params[:page] || 1
   end
 
+  def associate_root_article(associated_article_id)
+    article = portal.articles.find(associated_article_id) if associated_article_id.present?
```

Comment:
We might not need to do recursion with the query.

we could simplify the logic to 

- we fetch the associated article. 
- attach it to current articles if the associated article doesn't have another association ... 
- in the case of another association, we save that id instead.

this pattern would ensure that there won't be a deeper level association saved in the database.

Reply:
@sojan-official Makes sense if we are already associated root article from the start to the current article.